### DISCUSSION TITLE
[General] Issues creating LP pools on Testnet

### DISCUSSION THREAD

**Discussion by shane98x** [zkSync-Community-Hub/zksync-developers#338] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I'm trying to create the effect of making a pool of tokenA/tokenB and adding liquidity through the SyncSwap core contracts which is open sourced on their github: https://github.com/syncswap/core-contracts

### What is the specific issue or error you're encountering?

I'm always having my transactions fail f.e when trying to create a pool and/or add liquidity.

### Can you share the error messages or logs you're receiving, if any?

When trying to make a pool with the router contract: Error function_selector = 0x, data = 0x
Besides this error, I also get this one a lot (I thought unsufficient approvals caused this, yet even when making sure the allowance is there, it still throws this error): 
- There was an error while performing method call
- cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={"reason":"execution reverted","code":"UNPREDICTABLE_GAS_LIMIT","method":"estimateGas","transaction"...

### Have you made any recent changes to the contract before encountering this issue?

no, complete copy paste and some naming changes of the contract itself

### Are there any external libraries or contracts that your contract interacts with?

basically all contracts stored in the lib file of the core syncswap repo

### Can you provide the relevant portions of your contract code where the issue is occurring?

/// @notice Wrapper function to allow pool deployment to be batched.
    function createPool(address _factory, bytes calldata data) external payable returns (address) {
        return IPoolFactory(_factory).createPool(data);
    }

--> it happens almost everywhere though

### Have you tried to isolate the problem, and if so, what were the results?

I can use other functions within the smart contract to read but almost every write actions results in the errors described above

### What steps have you already taken to try to resolve the issue?

try to write transactions throught the zksync cli : npx zksync-cli contract write instead of the explorer manually

### Repo Link (Optional)

https://github.com/syncswap/core-contracts

**Comment by dutterbutter** [comment, marked as answer]
Hey @shane98x 

Its probably better to open this issue with the syncswap team in the repo as we are not the most familiar with their implementation.

**Comment by velezlia** [comment]
It seems like there might be issues with the transaction execution or gas estimation process. I would recommend checking the contract configurations, ensuring proper function parameters, and verifying that the required allowances and balances are set correctly. Additionally, debugging through the contract code and closely examining the error messages received during transaction failures could provide insights into the root cause of the issue.

**Reply by shane98x** [reply]
I fixed it by creating my own byte data, still have issues inputting tuple values

**Comment by jecgaegacharsmi** [comment]
Encountering errors while interacting with SyncSwap core contracts on the testnet. Transactions fail when attempting to create a pool or add liquidity. Even with sufficient approvals, transactions still revert unexpectedly. Seeking assistance to resolve these issues.

**Reply by shane98x** [reply]
Make sure the byte data is in the correct format

**Comment by hunastly** [comment]
Troubleshooting SyncSwap Core Contracts on Testnet.

**Comment by mapeshewa** [comment]
The user is trying to create a pool of tokenA/tokenB and add liquidity through the SyncSwap core contracts on Testnet. However, they are encountering errors such as "Error function_selector = 0x, data = 0x" and "cannot estimate gas; transaction may fail or may require manual gas limit". The user has not made any recent changes to the contract and is interacting with external libraries and contracts stored in the lib file of the core SyncSwap repository. They have tried using other functions within the smart contract to read, but almost every write action results in the aforementioned errors. The user has also tried writing transactions through the zksync cli instead of the explorer manually.
